src/tinygp/kernels/quasisep.py

```diff
@@ -635,134 +638,362 @@ def transition_matrix(self, X1: JAXArray, X2: JAXArray) -> JAXArray:
+                in the definition above. This should be an array of length `q+1`,
+                where `q+1 <= p`.
         """
-        sigma = jnp.ones(()) if sigma is None else sigma
+        sigma = jnp.ones(())
         alpha = jnp.atleast_1d(alpha)
         beta = jnp.atleast_1d(beta)
         assert alpha.ndim == 1
         assert beta.ndim == 1
         p = alpha.shape[0]
         assert beta.shape[0] <= p
 
-        # We find the roots of the autoregressive polynomial as a means to find
-        # the eigendecomposition of the design matrix.
-        alpha_ext = jnp.append(alpha, 1.0)
-        roots = jnp.roots(alpha_ext[::-1], strip_zeros=False)
-        proj = roots[:, None] ** jnp.arange(p)[None, :]
-        proj_inv = jnp.linalg.inv(proj)
-
-        # Compute the stationary covariance - there is almost certainly a more
-        # elegant way, but this works! I worked this out kind of by trial and
-        # error using sympy. There is a lot of known structure in the P_inf
-        # matrix that can be exploited to "simplify" this calculation.
-        # Specifically, there are only `p` degrees of freedom, and P_inf has the
-        # following structure:
-        #
-        #   P_inf = [
-        #     [ p0   0   -p1   0    p2 ]
-        #     [ 0    p1   0   -p2   0  ]
-        #     [-p1   0    p2   0   -p3 ]
-        #     [ 0   -p2   0    p3   0  ]
-        #     [ p2   0   -p3   0    p4 ]
-        #   ]
-        #
-        # Using this structure, we get can solve the usual:
-        #
-        #  A @ P + P @ A.T + L @ L.T = 0
-        #
-        # for `P`, and we get something like the following. Kelly et al. (2104)
-        # also have an expression for this (their V_{ij}), but I prefer to use
-        # this since it is probably roughly just as fast to compute, and it is
-        # strictly real-valued.
-        f = 2 * ((np.arange(2 * p) // 2) % 2) - 1
-        x = f * jnp.append(alpha_ext, jnp.zeros(p - 1))
-        params = jnp.stack([jnp.roll(x, k)[::2] for k in range(p)], axis=0)
-        params = jnp.linalg.solve(params, 0.5 * sigma**2 * jnp.eye(p, 1, k=-p + 1))[
-            :, 0
-        ]
-        stn_ = []
-        for j in range(p):
-            stn_.append([jnp.zeros(()) for _ in range(p)])
-            for n, k in enumerate(range(j - 2, -1, -2)):
-                stn_[-1][k] = (2 * (n % 2) - 1) * params[j - n - 1]
-            for n, k in enumerate(range(j, p, 2)):
-                stn_[-1][k] = (1 - 2 * (n % 2)) * params[n + j]
-        stn = jnp.array(list(map(jnp.stack, stn_)))
+        # Find acvf using Eqn. 4 in Kelly+14, giving the correct combination of
+        # real/complex exponential kernels
+        arroots = carma_roots(jnp.append(alpha, 1.0))
+        acf = carma_acvf(arroots, alpha, beta * sigma)
+
+        # Mask for real/complex exponential kernels
+        _real_mask = jnp.abs(arroots.imag) < 10 * jnp.finfo(arroots.imag.dtype).eps
```

Comment:
I changed this from `== 0.0` since comparing floating point numbers like that is always risky!

Reply:
Great, this is safer for sure!